src/ObjectUtils/PdbFileLlvm.cpp

```diff
@@ -26,8 +31,14 @@ namespace {
 // all symbol info required for functions.
 class SymbolInfoVisitor : public llvm::codeview::SymbolVisitorCallbacks {
  public:
-  SymbolInfoVisitor(ModuleSymbols* module_symbols, const ObjectFileInfo& object_file_info)
-      : module_symbols_(module_symbols), object_file_info_(object_file_info) {}
+  SymbolInfoVisitor(ModuleSymbols* module_symbols, const ObjectFileInfo& object_file_info,
+                    llvm::pdb::TpiStream* type_info_stream)
+      : module_symbols_(module_symbols),
+        object_file_info_(object_file_info),
+        type_info_stream_(type_info_stream) {
+    ORBIT_CHECK(module_symbols != nullptr);
```

Comment:
Nit: A bit undecided what is better here, but I generally prefer to have CHECKs close to where they matter (locality). In this case, in the code / function where the pointers are actually dereferenced.

Reply:
I agree. However, I also see the assignment here as "usage", where we want to fail if someone is using the class unintended.

I added additional checks at the usages inside the functions.

Reply:
> I see the CHECKS also more as an explicit statement about the expectations about the calls. So in this case, expressing that we don't support constructing this class with nullptrs.

+1

Reply:
yes